modopt/opt/gradient.py

```diff
@@ -55,12 +57,13 @@ class GradParent(object):
     """
 
     def __init__(self, data, op, trans_op, get_grad=None, cost=None,
-                 data_type=None):
+                 data_type=None, verbose=True):
 
         self._grad_data_type = data_type
         self.obs_data = data
         self.op = op
         self.trans_op = trans_op
+        self.verbose = verbose
```

Comment:
Sorry, one more thing. You need to move this definition above `self.obs_data = data` otherwise `self.verbose` is not defined when the data is allocated. I suggest simply moving this definition to the top.

i.e.

```python
self.verbose = verbose
self._grad_data_type = data_type
self.obs_data = data
self.op = op
self.trans_op = trans_op    
```

Reply:
Oh yes, its true. I just pushed a new commit fixing it.